token/program-2022-test/tests/confidential_transfer.rs

```diff
@@ -1237,43 +1229,16 @@ async fn ct_transfer() {
         )
         .await;
 
-    let state = token
-        .get_account_info(&alice_meta.token_account)
-        .await
-        .unwrap();
-    let extension = state
-        .get_extension::<ConfidentialTransferAccount>()
-        .unwrap();
-
-    let err = token
-        .confidential_transfer_transfer(
-            &alice_meta.token_account,
+    token
+        .confidential_transfer_apply_pending_balance(
             &alice_meta.token_account,
-            &alice,
-            0, // amount
-            0,
-            &extension.available_balance.try_into().unwrap(),
-            &alice_meta.elgamal_keypair.public,
-            Some(ct_mint_transfer_auditor_elgamal_keypair.public),
+            &alice.pubkey(),
+            None,
+            alice_meta.elgamal_keypair.secret(),
+            &alice_meta.aes_key,
+            &[&alice],
         )
         .await
-        .unwrap_err();
-
-    assert_eq!(
-        err,
-        TokenClientError::Client(Box::new(TransportError::TransactionError(
-            TransactionError::InstructionError(
-                0,
-                InstructionError::Custom(
-                    TokenError::MaximumPendingBalanceCreditCounterExceeded as u32
-                ),
-            )
-        )))
-    );
```

Comment:
This seems like an important case to check, can you add it back in?

Reply:
This check is verifying that a confidential transfer fails if the maximum credit limit in the destination account is reached. In practice the default destination will be `2^16`, but in order to test this, we need to set the destination account maximum credit limit to be low (say `2`). 

In order to simplify the tests, to initialize the `alice_meta`, I used the helper constructor  `ConfidentialTokenAccountMeta::new_with_tokens`, which deposits some funds to `alice`'s account. The `new_with_tokens` interface unfortunately does not allow us to set custom maximum credit limit. Instead of adding a new constructor `ConfidentialTokenAccountMeta::new_with_tokens_and_maximum_pending_balance_credit_counter`, I just initialize `bob_meta` with the constructor `ConfidentialTokenAccountMeta::new_with_maximum_pending_balance_credit_counter` and tested that a transfer to `bob`'s account fails if the maximum credit counter is reached.

So in summary, this check on `alice`'s account is not removed, but it is just moved to work on `bob`'s account purely instead for simplicity.